### PR TITLE
fix: manage Marked for Execution debuff from Slayer's Strike

### PR DESCRIPTION
**Pull request by johnnylam88**
Modify Arms and Fury warrior modules to manage the Marked for Execution stacking debuff from Slayer's Strike. Use the correct name for the debuff (also used in SimC) as `marked_for_execution` and allow it to be tracked via `active_dot`.

### CODE REVIEW

File: TheWarWithin/WarriorArms.lua [hunk 1]
```diff
@@ -590,16 +600,32 @@ local TriggerCollateralDamage = setfenv( function()
...
+
 spec:RegisterCombatLogEvent( function( _, subtype, _,  sourceGUID, sourceName, _, _, destGUID, destName, destFlags, _, spellID, spellName, _, _, _, _, critical_swing, _, _, critical_spell )
     if sourceGUID == state.GUID then
         if subtype == "SPELL_CAST_SUCCESS" then
             if ( spellName == class.abilities.colossus_smash.name or spellName == class.abilities.warbreaker.name ) then
                 last_cs_target = destGUID
             end
+        elseif subtype == "SPELL_DAMAGE" and UnitGUID( "target" ) == destGUID then
+            if spellID == 445579 then -- Slayer's Strike occurred
+                marked_for_execution_stacks[ destGUID ] = max( ( marked_for_execution_stacks[ destGUID ] or 0 ) + 1, 3 )
```
Comment (Hekili) [inline TheWarWithin/WarriorArms.lua]: Should probably be `min` if capped at 3.

File: TheWarWithin/WarriorFury.lua [hunk 2]
```diff
@@ -1303,14 +1303,15 @@ spec:RegisterAbilities( {
         indicator = function () if cycle_for_execute then return "cycle" end end,
 
         handler = function ()
-            if talent.imminent_demise.enabled then
-                addStack( "imminent_demise" )
-            end
+            removeBuff( "marked_for_execution" )
```
Comment (Hekili) [inline TheWarWithin/WarriorFury.lua]: `removeDebuff` ?

File: TheWarWithin/WarriorArms.lua [hunk 3]
```diff
@@ -1052,6 +1089,7 @@ spec:RegisterAbilities( {
             return rage.time_to_20
         end,
         handler = function ()
+            removeBuff( "marked_for_execution" )
```
Comment (Hekili) [inline TheWarWithin/WarriorArms.lua]: `removeDebuff` ?

File: TheWarWithin/WarriorArms.lua (hunk 1 above)
Comment (Hekili) [reply TheWarWithin/WarriorArms.lua]: LMK if you agree/disagree; I'm pretty sure we want `min` or it's going to apply 3 stacks on the first cast and then count up from there.

**Review by Hekili — APPROVED** [review]

**Comment by johnnylam88** [comment]
Sorry, missed all your comments for some reason. My notifications didn't work. You were right on the min/max thing.

**Pull request merged**